ntapfuse_ops.c

```diff
@@ -0,0 +1,375 @@
+  //log the results of the write 
+  log_write(ret,buf,path);
+  return ret; 
+  // return pwrite (fi->fh, buf, size, off) < 0 ? -errno : size;
+}
+
+int
+ntapfuse_statfs (const char *path, struct statvfs *buf)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return statvfs (fpath, buf) ? -errno : 0;
+}
+
+int
+ntapfuse_release (const char *path, struct fuse_file_info *fi)
+{
+  return close (fi->fh) ? -errno : 0;
+}
+
+int
+ntapfuse_fsync (const char *path, int datasync, struct fuse_file_info *fi)
+{
+  if (datasync)
+    return fdatasync (fi->fh) ? -errno : 0;
+  else
+    return fsync (fi->fh) ? -errno : 0;
+}
+
+int
+ntapfuse_setxattr (const char *path, const char *name, const char *value,
+	       size_t size, int flags)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return lsetxattr (fpath, name, value, size, flags) ? -errno : 0;
+}
+
+int
+ntapfuse_getxattr (const char *path, const char *name, char *value, size_t size)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  ssize_t s = lgetxattr (fpath, name, value, size);
+  return s < 0 ? -errno : s;
+}
+
+int
+ntapfuse_listxattr (const char *path, char *list, size_t size)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return llistxattr (fpath, list, size) < 0 ? -errno : 0;
+}
+
+int
+ntapfuse_removexattr (const char *path, const char *name)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return lremovexattr (fpath, name) ? -errno : 0;
+}
+
+int
+ntapfuse_opendir (const char *path, struct fuse_file_info *fi)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  DIR *dir = opendir (fpath);
+  if (dir == NULL)
+    return -errno;
+
+  fi->fh = (uint64_t) dir;
+
+  return 0;
+}
+
+int
+ntapfuse_readdir (const char *path, void *buf, fuse_fill_dir_t fill, off_t off,
+	      struct fuse_file_info *fi)
+{
+  struct dirent *de = NULL;
+
+  while ((de = readdir ((DIR *) fi->fh)) != NULL)
+    {
+      struct stat st;
+      memset (&st, 0, sizeof (struct stat));
+      st.st_ino = de->d_ino;
+      st.st_mode = de->d_type << 12;
+
+      if (fill (buf, de->d_name, &st, 0))
+	break;
+    }
+
+  return 0;
+}
+
+int
+ntapfuse_releasedir (const char *path, struct fuse_file_info *fi)
+{
+  return closedir ((DIR *) fi->fh) ? -errno : 0;
+}
+
+int
+ntapfuse_access (const char *path, int mode)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return access (fpath, mode) ? -errno : 0;
+}
+
+void *
+ntapfuse_init (struct fuse_conn_info *conn)
+{
+  return (fuse_get_context())->private_data;
+}
+
+/*
+  Creates a file (if one doesn't already exist) named log.txt 
+  in the source directory where the executable is located and 
+  writes information regarding an attempted write:
+    -if write was successsful
+    -size of the write
+    -contents desired to be written
+    -date of the write
+    -path of the write 
+*/
+void log_write(int ret, const char* buf, const char *path)
```

Comment:
I see that from another pull request, there is a logger.h and cc file were added. Consider moving any logging code into that file.